bgpd/bgp_vty.c

```diff
@@ -10717,6 +10717,16 @@ static int bgp_show_neighbor(struct vty *vty, struct bgp *bgp,
 				}
 			}
 			break;
+		case show_ipv4_peer:
+			if (BGP_PEER_SU_IPV4(peer)) {
```

Comment:
The idea behind your change is correct and fixes an issue, but this is the wrong thing to check here. `show bgp ipv4` refers to the AFI of the routes, not to the AF of the BGP session.

If you configure a BGP session over IPv4 and enable the IPv6 address family, it needs to show up in `show bgp ipv6` too because it carries IPv6 routes.  This especially affects RFC5549 BGP sessions which run over IPv6 but carry IPv4 routes (with IPv6 nexthops.)

=> You need to check `peer->afc[afi][safi]` instead.

Reply:
Agree. Thanks. That's what i understood after looking at the documentation http://docs.frrouting.org/en/latest/bgp.html#clicmd-showbgp[afi][safi]. Need to check peer->afc[afi][safi]. Will resubmit the fixes.